packages/e2e/lib/e2e_perf.dart

```diff
@@ -0,0 +1,203 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ui';
+
+import 'package:flutter/scheduler.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter/widgets.dart';
+
+import 'package:e2e/e2e.dart';
+
+/// The maximum amount of time considered safe to spend for a frame's build
+/// phase. Anything past that is in the danger of missing the frame as 60FPS.
+///
+/// Changing this doesn't re-evaluate existing summary.
+Duration kBuildBudget = const Duration(milliseconds: 16);
+// TODO(CareF): Automatically calculate the refresh budget (#61958)
+
+/// Usually it's recommended to limit callbacks of the test to [WidgetController]
+/// API so it can be more universally used.
+typedef ControlCallback = Future<void> Function(WidgetController controller);
```

Comment:
This is not used in this change, do you have following up PR which takes this up? It might be better to move it there.

Reply:
Removed. This typedef will be widely used in a bunch of performance test cases in the flutter/flutter repo. I should find a better place for its definition.